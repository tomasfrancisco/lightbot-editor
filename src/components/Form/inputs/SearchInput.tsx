import { Input } from "antd";
import * as React from "react";
import styled from "react-emotion";

const Search = Input.Search;

const SearchContainer = styled("div")`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

type SearchInputProps = {
  placeholder?: string;
  enterButton?: string;
  onSearchResult: (value: string) => void;
};

export class SearchInput extends React.Component<SearchInputProps> {
  public render() {
    const { placeholder, enterButton } = this.props;
    return (
      <SearchContainer>
        <Search
          placeholder={placeholder || "Search here"}
          enterButton={enterButton}
          onSearch={this.props.onSearchResult}
          onChange={this.onChange}
        />
      </SearchContainer>
    );
  }

  private onChange = event => {
    this.props.onSearchResult(event.target.value);
  };
}
