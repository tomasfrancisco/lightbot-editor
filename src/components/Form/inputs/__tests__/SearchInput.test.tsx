import { shallow } from "enzyme";
import * as React from "react";

import { SearchInput } from "../SearchInput";

describe("components/Form/inputs/TitleInput", () => {
  test("renders", () => {
    const onSearch = value => {
      /**/
    };
    const tree = shallow(<SearchInput onSearchResult={onSearch} />);
    expect(tree).toMatchSnapshot();
  });
});
