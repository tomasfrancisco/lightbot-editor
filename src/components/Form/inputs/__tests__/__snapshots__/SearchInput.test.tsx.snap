// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/inputs/TitleInput renders 1`] = `
<ForwardRef(render)>
  <Search
    enterButton={false}
    onChange={[Function]}
    onSearch={[Function]}
    placeholder="Search here"
  />
</ForwardRef(render)>
`;
